File: Week-06/sentimental-mario-less/mario.py
from cs50 import get_int


def main():
    while True:
        # Ask user for the height
        height = get_int("Height: ")

        # Move on if height is in between 1 and 8
        if height in range(1, 9):
            break

    # Store variables for spaces and hashes
    spaces = height
    hashes = 0

    # Loop through accepted value printing hashes and spaces
    for i in range(height):
        spaces -= 1
        hashes += 1
        print(" " * spaces, end="")
        print("#" * hashes)


if __name__ == "__main__":
    main()

File: Week-01/mario-less/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int height;
    int row;
    int spaces;
    int hashes;

    // Ask user for the height value
    do
    {
        height = get_int("Height: ");
    }

    // Accept only positive inputs from 1 to 8
    // If value is not accepted, ask again
    while (height < 1 || height > 8);

    // Loop through accepted value printing hashes and spaces
    for (row = 1; row <= height; row++)
    {
        for (spaces = (height - row); spaces > 0; spaces--)
        {
            printf(" ");
        }
        for (hashes = 1; hashes <= row; hashes++)
        {
            printf("#");
        }
        printf("\n");
    }
};
